compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -744,12 +743,12 @@ impl<'tcx> TyCtxt<'tcx> {
         self,
         value: Binder<'tcx, T>,
         mut fld_r: F,
-    ) -> (T, BTreeMap<ty::BoundRegion, ty::Region<'tcx>>)
+    ) -> (T, FxHashMap<ty::BoundRegion, ty::Region<'tcx>>)
```

Comment:
Is this FxHashMap used anyhere?

Reply:
Actually, it might not be! 👀

Reply:
Maybe not? 👀 🤔 I'm not entirely sure how to check. The `lsp-find-references` command in Emacs shows several places where the returned map is ignored:

```
compiler/rustc_borrowck/src/universal_regions.rs
736:         let (value, _map) = self.tcx.replace_late_bound_regions(value, |br| {
compiler/rustc_middle/src/ty/fold.rs
821:         self.replace_late_bound_regions(value, |br| {
911:         self.replace_late_bound_regions(value, |_| self.lifetimes.re_erased).0
928:             .replace_late_bound_regions(sig, |_| {
compiler/rustc_middle/src/ty/print/pretty.rs
2160:             self.tcx.replace_late_bound_regions(value.clone(), |br| {
compiler/rustc_typeck/src/collect.rs
457:                                             .replace_late_bound_regions(poly_trait_ref, |_| {
```